periodic sync upstream KF to midstream ODH

tarilabs opened the pull request:
- keep https://github.com/kubeflow/model-registry/pull/1268
- keep https://github.com/kubeflow/model-registry/pull/1242
- keep https://github.com/kubeflow/model-registry/pull/1269
- keep https://github.com/kubeflow/model-registry/pull/1271
- keep https://github.com/kubeflow/model-registry/pull/1261
- keep https://github.com/kubeflow/model-registry/pull/1219
- fd9778c3c2df4d6ba63eacd6b576f61490dff2a5 merge commit resolving conflict on .github/workflows/python-tests.yml
by adding the fuzz test step in job here in m/s

labels are applied

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **New Features**
  * Added support for "federated" deployment mode and UI theming via environment variables and Makefile targets.
  * Introduced an NGINX-based authentication proxy for standalone UI deployments.
  * Added fuzz testing capability for the Python client, including new Makefile targets, nox sessions, and test fixtures.
  * Enhanced documentation for new deployment modes, UI theming, and fuzz testing instructions.

* **Improvements**
  * Unified deployment mode configuration across UI, backend, and manifests, replacing legacy flags with a single `DEPLOYMENT_MODE`.
  * Improved error handling for non-JSON API responses in the UI backend.
  * Optimized Kubernetes namespace access checks with parallelization for better performance.
  * Updated service and deployment manifests to reflect new deployment modes and proxy integration.

* **Bug Fixes**
  * Corrected environment variable usage and conditional rendering logic in the UI for deployment modes.
  * Fixed test behavior to remove unnecessary expected-failure markers.

* **Chores**
  * Updated dependencies and cleaned up unused environment variables and code references.
  * Improved `.gitignore` and documentation for clarity.

* **Tests**
  * Added comprehensive fuzz and stateful property-based tests for the Python client.
  * Enhanced test cleanup and resource management for better isolation and reliability.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

isinyaaa commented:
@dtrifiro do you think this should work or would you rather go with `fetch-depth: 0` to get all history + tags?

dtrifiro commented:
I think it would be ok.

What about explicitly using:

```yaml
on:
  release:
    types: [published]
```

and explicitly creating the release (and release notes) on github?

isinyaaa commented:
Updated and tested on: https://github.com/isinyaaa/model-registry/actions/runs/7197484144/job/19604711876

adysenrothman committed:
update broken link for dir rename in manifests (#1268)

Signed-off-by: Adysen Rothman <85646824+adysenrothman@users.noreply.github.com>

fege committed:
Add api tests stateless (#1242)

* Add api tests stateless

Signed-off-by: fege <fmosca@redhat.com>

* Fix github action failures, lint and nox

Signed-off-by: fege <fmosca@redhat.com>

* Add too_slow and remove code to force the model version id

Signed-off-by: fege <fmosca@redhat.com>

* Add filter_too_much

Signed-off-by: fege <fmosca@redhat.com>

* add hook to avoid Unsatisfiable

Signed-off-by: fege <fmosca@redhat.com>

* remove variable

Signed-off-by: fege <fmosca@redhat.com>

* add artifact_states

Signed-off-by: fege <fmosca@redhat.com>

* Add example to schema

Signed-off-by: fege <fmosca@redhat.com>

* Add example in src

Signed-off-by: fege <fmosca@redhat.com>

* register strategy for string of int64

Signed-off-by: fege <fmosca@redhat.com>

* sort imports

Signed-off-by: fege <fmosca@redhat.com>

* modify case for problematic endpoints

Signed-off-by: fege <fmosca@redhat.com>

* add more examples

Signed-off-by: fege <fmosca@redhat.com>

* pin urllib

Signed-off-by: fege <fmosca@redhat.com>

* exclude problematic endpoints and test with valid data

Signed-off-by: fege <fmosca@redhat.com>

* Add gha to run the test and mark them with fuzz

Signed-off-by: fege <fmosca@redhat.com>

* revert change in Makefile pushed by error

Signed-off-by: fege <fmosca@redhat.com>

* skip test not marked with e2e or fuzz, trigger the fuzz on pr comment

Signed-off-by: fege <fmosca@redhat.com>

* trigger with label

Signed-off-by: fege <fmosca@redhat.com>

* correct the label name

Signed-off-by: fege <fmosca@redhat.com>

* add types and semplify the if

Signed-off-by: fege <fmosca@redhat.com>

* do not run e2e if test-fuzz label is added

Signed-off-by: fege <fmosca@redhat.com>

* unpin urllib

Signed-off-by: fege <fmosca@redhat.com>

* modify lock

Signed-off-by: fege <fmosca@redhat.com>

* fix: remove proc on label

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

---------

Signed-off-by: fege <fmosca@redhat.com>
Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>
Co-authored-by: Alessio Pragliola <seth.pro@gmail.com>

Al-Pragliola committed:
feat: reintroduce e2e order by test (#1269)

Signed-off-by: Alessio Pragliola <seth.pro@gmail.com>

fege committed:
Add state machine tests (#1271)

* Add state machine tests

Signed-off-by: fege <fmosca@redhat.com>

* adjust if to test on the open PR

Signed-off-by: fege <fmosca@redhat.com>

* revert if and remove debugging prints

Signed-off-by: fege <fmosca@redhat.com>

* revert if and remove debugging prints

Signed-off-by: fege <fmosca@redhat.com>

* Remove db-cleanup

Signed-off-by: fege <fmosca@redhat.com>

---------

Signed-off-by: fege <fmosca@redhat.com>

dependabot[bot] committed:
build(deps-dev): bump ts-jest from 29.3.4 to 29.4.0 in /clients/ui/frontend (#1261)

Bumps [ts-jest](https://github.com/kulshekhar/ts-jest) from 29.3.4 to 29.4.0.
- [Release notes](https://github.com/kulshekhar/ts-jest/releases)
- [Changelog](https://github.com/kulshekhar/ts-jest/blob/main/CHANGELOG.md)
- [Commits](https://github.com/kulshekhar/ts-jest/compare/v29.3.4...v29.4.0)

---
updated-dependencies:
- dependency-name: ts-jest
  dependency-version: 29.4.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

lucferbux committed:
Tweak to support federated platform (#1219)

Signed-off-by: lucferbux <lferrnan@redhat.com>

tarilabs committed:
Merge remote-tracking branch 'upstream/main' into tarilabs-20250704-sync

Resolved conflict in  .github/workflows/python-tests.yml
by adding the fuzz test step in job here in m/s

Signed-off-by: Matteo Mortari <matteo.mortari@gmail.com>
